solver/cachemanager.go

```diff
@@ -30,15 +30,25 @@ func NewCacheManager(ctx context.Context, id string, storage CacheKeyStorage, re
 		bklog.G(ctx).Errorf("failed to release unreferenced cache metadata: %+v", err)
 	}
 
+	go cm.pruneKeyMap(ctx)
+
 	return cm
 }
 
+type cacheKeyWithTime struct {
+	c *CacheKey
+	t time.Time
+}
+
 type cacheManager struct {
 	mu sync.RWMutex
 	id string
 
 	backend CacheKeyStorage
 	results CacheResultStorage
+
+	keys   map[string]cacheKeyWithTime
+	keysMu sync.Mutex
```

Comment:
is this mutex required? it looks like the calls to `newKeyWithID` are done by functions that already hold `mu`.

Reply:
It may be better to use a separate mutex as `mu` locks multiple longer-running operations that aren't related to the `keys` map.